client/src/components/TestSignUpForm.js

```diff
@@ -15,7 +15,7 @@ const TestSignUpForm = ({isAuth,signUp}) => {
         email: Yup.string().email("Invalid Email").required("Please enter your email"),
         password: Yup.string().required("password is required"),
         confirmPassword: Yup.string().when("password",{
-            is: val=> (!!(val && val.length)),
+            is: val=> (!!(val && val.length)), //double exclamation casts boolean

```

Comment:
!! (and Boolean("some value")) are generally not a good practice as it can introduce bugs in the long run, and also make the code less clean, try to refactor without it

Reply:
oooh ok noted boss! illl change it on the next commit